### PR TITLE
Haijian/merge v15 to master

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Don't Squash! Merge

This change is [Reviewable](https://reviewable.io/reviews/vaadin/flow-and-components-documentation/1177)

### CODE REVIEW

File: documentation/advanced/tutorial-switch-npm-pnpm.asciidoc [hunk 1]
```diff
@@ -1,40 +1,58 @@
...
 
 ifdef::env-github[:outfilesuffix: .asciidoc]
 
-= Differences Between pnpm and npm
+= Switch between NPM and PNPM
 
-Starting from Vaadin 14.2 and 15 _pnpm_ (AKA _performant npm_) tool is used as the default frontend package manager instead of _npm_.
-https://pnpm.js.org/[pnpm] uses a shared repository for all projects in a system which allows to reduce
-packages download time. On the contrast, by default npm will download all dependencies again for every project.
+Vaadin 15 uses `npm` as the default package manager. It's also possible to switch to https://pnpm.js.org/[pnpm], the benefit of `pnpm` is that it uses a shared repository, which allows reducing the build time.
```
Comment (pleku) [inline documentation/advanced/tutorial-switch-npm-pnpm.asciidoc]: This should be fixed and mention that "npm is the default package manager. Starting from 14.2 and 15, it is possible to switch to pnpm..."

File: documentation/advanced/tutorial-switch-npm-pnpm.asciidoc [hunk 2]
```diff
@@ -1,40 +1,58 @@
...
 
-If pnpm is not installed globally the framework will install it locally using npm.
```
Comment (pleku) [inline documentation/advanced/tutorial-switch-npm-pnpm.asciidoc]: This should not be removed in master now, as the feature will be in 3.1 to install it globally.

File: documentation/kk.tree [hunk 3]
```diff
@@ -0,0 +1,182 @@
+[INFO] Scanning for projects...
```
Comment (pleku) [inline documentation/kk.tree]: what is this ? remove ?

File: documentation/introduction/introduction-overview.asciidoc [hunk 4]
```diff
@@ -53,8 +79,15 @@ You can use the same Vaadin components in JavaScript as follows, with code equiv
...
 
+<<<<<<< HEAD
+See the tutorials in <<../polymer-templates/tutorial-template-basic#,Creating Polymer Templates>> for more.
+
+
+== Routing and Navigation in Java component using `@Router` annotation
+=======
 // Create the vaadin-text-field element
 const textField = document.createElement('vaadin-text-field');
+>>>>>>> master
```
Comment (pleku) [inline documentation/introduction/introduction-overview.asciidoc]: Merge conflict left unresolved properly

File: documentation/pom.xml [hunk 5]
```diff
@@ -54,16 +54,47 @@
...
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-client</artifactId>
+            <version>${flow.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.vaadin</groupId>
+            <artifactId>flow-html-components</artifactId>
+            <version>${flow.version}</version>
+        </dependency>
```
Comment (pleku) [inline documentation/pom.xml]: Why are these flow dependencies added ? These are already in vaadin and controlled by flow-bom ?

File: documentation/kk.tree (hunk 3 above)
Comment (haijian-vaadin) [reply documentation/kk.tree]: deleted

File: documentation/introduction/introduction-overview.asciidoc (hunk 4 above)
Comment (haijian-vaadin) [reply documentation/introduction/introduction-overview.asciidoc]: good catch, resolved

File: documentation/pom.xml (hunk 5 above)
Comment (haijian-vaadin) [reply documentation/pom.xml]: I guess some legacy stuff from V15 branch, now removed

File: documentation/advanced/tutorial-switch-npm-pnpm.asciidoc (hunk 2 above)
Comment (haijian-vaadin) [reply documentation/advanced/tutorial-switch-npm-pnpm.asciidoc]: added back

**Pull request merged**